qbft/decided.go

```diff
@@ -49,8 +51,9 @@ func (c *Controller) UponDecided(msg *SignedMessage) (*SignedMessage, error) {
 		i.State.Round = msg.Message.Round
 		i.State.Decided = true
 		i.State.DecidedValue = data.Data
-		c.StoredInstances.addNewInstance(i)
-
+		if err := c.GetConfig().GetStorage().SaveInstanceState(i.State); err != nil {
```

Comment:
could cause a case where we decided current instance but saving failed

Reply:
thats why returning error

Reply:
yeah but you decide the instance before

Reply:
any way if there is an error it wont be saved.. 
we can in case of error revert the instance changes to be not decided for example 
WDYT

Reply:
should return error